tools/java_api/KuzuNodeValue.java

```diff
@@ -0,0 +1,73 @@
+package tools.java_api;
+
+public class KuzuNodeValue {
+    long nv_ref;
+    boolean destroyed = false;
+    boolean isOwnedByCPP = false;
+
+    private void checkNotdestroyed() {
+        assert !destroyed: "KuzuNodeValue has been destoryed.";
```

Comment:
`assert` is mainly used for internal error handling. Throw an exception instead.